### PR TITLE
diagnostic: fix index out of bounds error in sysinfo

### LINKED ISSUE #15006: diagnostic: non-stop cpu expansion causes index out of bounds error
### What version of TiKV are you using?

`master`

### What operating system and CPU are you using?

```
Linux x86_64/aarch64
```

### Steps to reproduce

1. Boot a TiDB cluster.

2. Execute the following query:

```sql
SELECT * FROM information_schema.cluster_hardware;
```

3. Expand the number of TiKV host's CPU cores without restarting TiKV process.

4. Re-execute the query in step-2.

Then TiKV crashed.

If step-3 is not easy to achieve, we can simulate it in the following way:

1. Copy `/proc/stat`:
```shell
$ cp /proc/stat /tmp/proc/stat
```

2. Edit `sysinfo/src/linux/cpu.rs`, temporarily use `/tmp/proc/stat` instead of `/proc/stat`:

```rust
pub(crate) fn refresh(&mut self, only_update_global_cpu: bool, refresh_kind: CpuRefreshKind) {
  // ...
  if need_cpu_usage_update {
      self.last_update = Some(Instant::now());
      let f = match File::open("/proc/stat") { // "/proc/stat" -> "/tmp/proc/stat"
          Ok(f) => f,
          Err(_e) => {
              sysinfo_debug!("failed to retrieve CPU information: {:?}", _e);
              return;
          }
      };
  // ...
```

3. Rebuild and restart TiKV, then re-run `SELECT * FROM information_schema.cluster_hardware;`.

4. Edit `/tmp/proc/stat`, add one CPU core:

```
cpu  888202 0 64366 33777873 6620 0 9253 0 0 0
cpu0 84111 0 6925 3369197 309 0 7126 0 0 0
cpu1 99737 0 8030 3364200 1975 0 1161 0 0 0
cpu2 102944 0 6564 3365370 338 0 254 0 0 0
cpu3 80441 0 6131 3387710 943 0 145 0 0 0
cpu4 79461 0 6371 3388871 531 0 94 0 0 0   <===== added line
...
```

5. Re-run `SELECT * FROM information_schema.cluster_hardware;`.

### What did you expect?

All going well.

### What did happened?

TiKV crashed:

```
[2023/06/27 00:18:15.154 +08:00] [FATAL] [lib.rs:497] ["index out of bounds: the len is 4 but the index is 4"] [backtrace="   0: tikv_util::set_panic_hook::{{closure}}\n   1: std::panicking::rust_panic_with_hook\n   2: std::panicking::begin_panic_handler::{{closure}}\n   3: std::sys_common::backtrace::__rust_end_short_backtrace\n   4: rust_begin_unwind\n   5: core::panicking::panic_fmt\n   6: core::panicking::panic_bounds_check\n   7: sysinfo::linux::cpu::CpusWrapper::refresh\n   8: tikv::server::service::diagnostics::sys::hardware_info\n   9: tokio::runtime::task::core::CoreStage<T>::poll\n  10: tokio::runtime::task::raw::poll\n  11: tokio::runtime::scheduler::multi_thread::worker::Context::run_task\n  12: tokio::runtime::scheduler::multi_thread::worker::run\n  13: tokio::runtime::task::raw::poll\n  14: std::sys_common::backtrace::__rust_begin_short_backtrace\n  15: core::ops::function::FnOnce::call_once{{vtable.shim}}\n  16: std::sys::unix::thread::Thread::new::thread_start\n  17: <unknown>\n  18: <unknown>\n"]
```

### PR DESCRIPTION
**Pull request by mornyx**
### What is changed and how it works?

Issue Number: Close #15006

What's Changed:

Just expand `cpus` when it's length is insufficient. This prevents `sysinfo` from force reusing fixed-length `cpus` which cause crashes. 

Related commit: https://github.com/tikv/sysinfo/commit/5a1bcf08816979624ef2ad79cfb896de432a9501

```commit-message
fix index out of bounds error in sysinfo.

* update sysinfo version to the personal branch
```

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- No code in TiKV
- All tests in `sysinfo` have passed

### Release note 

```release-note
None
```

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -193,6 +193,8 @@ fs2 = { git = "https://github.com/tabokie/fs2-rs", branch = "tikv" }
 # Remove this when a new version is release. We need to solve rust-lang/cmake-rs#143.
 cmake = { git = "https://github.com/rust-lang/cmake-rs" }
 
+sysinfo ={ git = "https://github.com/mornyx/sysinfo", branch = "0.26-fix-cpu" }
```
Comment (zhongzc) [inline Cargo.toml:196]: Have you attempted to fix it on https://github.com/tikv/sysinfo?

Comment (zhongzc) [reply Cargo.toml:196]: I prefer not to patch with a personal repo.

Comment (mornyx) [reply Cargo.toml:196]: That makes sense. I'm going to ask for `tikv/sysinfo` permissions.

**Review by zhongzc — APPROVED** [review]

**Review by CalvinNeo — APPROVED** [review]
lgtm

**Review by innerr — APPROVED** [review]
LGTM

**Review by kaixu120811 — COMMENTED** [review]
LGTM

**Pull request merged**